bgpd/bgpd.c

```diff
@@ -3222,8 +3222,21 @@ int bgp_get(struct bgp **bgp_val, as_t *as, const char *name,
 	}
 
 	bgp = bgp_create(as, name, inst_type);
-	if (bgp_option_check(BGP_OPT_NO_ZEBRA) && name)
-		bgp->vrf_id = vrf_generate_id();
+
+	/*
+	 * view instances will never work inside of a vrf
+	 * as such they must always be in the VRF_DEFAULT
+	 * Also we must set this to something useful because
+	 * of the vrf socket code needing an actual useful
+	 * default value to send to the underlying OS.
+	 *
+	 * This code is currently ignoring vrf based
+	 * code using the -Z option( and that is probably
+	 * best addressed elsewhere in the code )
+	 */
+	if (inst_type == BGP_INSTANCE_TYPE_VIEW)
+		bgp->vrf_id = VRF_DEFAULT;
+
```

Comment:
I wonder if it is really necessary to keep -Z option.
callocating bgp instance will set vrf_id to 0 aka VRF_DEFAULT.

in those conditions, BGP_OPT_NO_ZEBRA code could be removed too.

Reply:
Maybe, but this is just a backport to 7.5 to fix an issue with IPv6 views.

Reply:
`bgp_create` not only callocs the instance, but also sets vrf_id to VRF_UNKNOWN, so this code is needed.
I agree with removing `-Z` option, it is not needed anymore, but this change should be done on master, so please submit a PR if you want to do this.